### DISCUSSION TITLE
Java Agent flag OTEL_JAVAAGENT_ENABLED

### DISCUSSION THREAD

**Discussion by itsmaneesh** [open-telemetry/opentelemetry-java-instrumentation#5771] [Q&A]
Hi,
I was trying to use the env variable OTEL_JAVAAGENT_ENABLED with value as "false" to ensure that agent will not be used / invoked for auto-instrumentation,. but still agent is being invoked during service start up.
Can you please let me know if I am missing something here?

Thanks,
Maneesh

**Comment by mateuszrzeszutek** [comment]
Hey @itsmaneesh ,
The agent will always be invoked as long as you have the `-javaagent` parameter. When you set the `OTEL_JAVAAGENT_ENABLED` to `false`, it is _effectively_ disabled -- it will not perform any bytecode modifications, and will not generate any telemetry.

**Reply by trask** [reply]
@itsmaneesh what is the sign you are seeing of the javaagent being invoked? if it's logging something, maybe we could turn that off that logging when OTEL_JAVAAGENT_ENABLED is false
